x-pack/plugins/maps/public/classes/styles/vector/properties/dynamic_text_property.js

```diff
@@ -10,7 +10,11 @@ import { getComputedFieldName } from '../style_util';
 export class DynamicTextProperty extends DynamicStyleProperty {
   syncTextFieldWithMb(mbLayerId, mbMap) {
     if (this._field && this._field.isValid()) {
-      const targetName = getComputedFieldName(this._styleName, this._options.field.name);
+      // Fields that support auto-domain are normalized with a field-formatter and stored into a computed-field
+      // Otherwise, the raw value is just carried over and no computed field is created.
+      const targetName = this._field.supportsAutoDomain()
+        ? getComputedFieldName(this._styleName, this._field.getName())
```

Comment:
instead of `this._field.getName()`, use `this.getFieldName()`. `this._field` can be null and `this.getFieldName` already has logic in place to check for null fields.

Reply:
making the change for consistency, but `_field` _has_ to be valid, and this is enforced higher-up.